### PR TITLE
feat: playlist page and functionality

### PR DESCRIPTION
**Pull request by PuneetP16**
Features Added:
1. Create Playlist,
2. Add to Playlist,
3. Delete Playlist,
4. Remove Playlist,
5. Listing all Playlists and Listing all videos in Single Playlist.

Playlist Panel:
image

Create Playlist:
image

Add to Playlist 
image

Playlist Page
image

Single Playlist Page
image

### CODE REVIEW

File: src/components/Cards/PlaylistCard/PlaylistCard.css [hunk 1]
```diff
@@ -0,0 +1,33 @@
+.category_card__list {
+	position: relative;
+}
+
+.catergories__name {
+	/* background-color: var(--COLOR-PRIMARY); */
```
Comment (saurabhchirde) [inline src/components/Cards/PlaylistCard/PlaylistCard.css:6]: remove comments

File: src/components/Cards/PlaylistCard/PlaylistCard.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { useVideo } from "../../../contexts";
+import { useAxios } from "../../../customHooks";
+import "./PlaylistCard.css";
+import { bxIcons } from "../../../data/icons";
+
+export const PlaylistCard = ({ playlist }) => {
+	const { videoDispatch, getPlaylist, playlists, removeFromPlaylists } =
+		useVideo();
+	const { axiosRequest } = useAxios();
+	console.log(playlist.videos);
```
Comment (saurabhchirde) [inline src/components/Cards/PlaylistCard/PlaylistCard.jsx:12]: remove console log

File: src/pages/Playlists/Playlists.css [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	gap: 2rem;
+	/* padding-left: 2rem; */
+}
+
+.video__items {
+	display: flex;
+	flex-flow: row wrap;
+	gap: 2rem;
+	margin-bottom: 10rem;
+}
```
Comment (saurabhchirde) [inline src/pages/Playlists/Playlists.css:74]: "display: flex" is repeated, you can combine them to reduce the repetitions of code.

File: src/pages/SingleVideo/SingleVideo.js [hunk 4]
```diff
@@ -29,148 +48,169 @@ export const SingleVideo = () => {
...
+	// 		contentDetails: { duration },
+	// 		snippet: {
+	// 			channelTitle,
+	// 			description,
+	// 			localized: { title },
+	// 			publishedAt,
+	// 		},
+	// 		statistics: { viewCount },
+	// 	} = singleVideo;
+	// }
```
Comment (saurabhchirde) [inline src/pages/SingleVideo/SingleVideo.js:73]: remove commented code

File: src/pages/SingleVideo/SingleVideo.js [hunk 5]
```diff
@@ -29,148 +48,169 @@ export const SingleVideo = () => {
...
+												title="Playlist"
+												onClick={togglePlaylistPanel}
+											>
+												{bxIcons.playlist}
+											</button>
+										</div>
+									</section>
+								</div>
+								<h2 className="video_channel__title">{channelTitle}</h2>
+							</section>
```
Comment (saurabhchirde) [inline src/pages/SingleVideo/SingleVideo.js:185]: I guess your code is not formatted, it has too much spacings, try to use prettier

**Review by saurabhchirde — APPROVED** [review]
Good work on the UI and consistent code practices. 
Gave some minor suggestions, otherwise, you are good to merge.

File: src/pages/Playlists/Playlists.css (hunk 3 above)
Comment (PuneetP16) [reply src/pages/Playlists/Playlists.css:74]: Will do that

File: src/pages/SingleVideo/SingleVideo.js (hunk 4 above)
Comment (PuneetP16) [reply src/pages/SingleVideo/SingleVideo.js:73]: Will do that

File: src/components/Cards/PlaylistCard/PlaylistCard.jsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+	return (
+		<li
+			key={playlist._id}
+			className="categories__list category_card__list playlist__card"
+		>
+			<Link
+				to={`/playlists/${playlist._id}`}
+				// onClick={getPlaylistHandler}
```
Comment (dev-ashishsethi) [inline src/components/Cards/PlaylistCard/PlaylistCard.jsx:21]: do not comment code

File: src/contexts/videoContext.js [hunk 7]
```diff
@@ -220,6 +238,102 @@ export const VideoProvider = ({ children }) => {
...
+	const getPlaylist = async (axiosRequest, playlistId) => {
+		console.log("from asycn", { playlistId });
+		try {
+			const playlistsURL = `/api/user/playlists/${playlistId}`;
+			const { output } = await axiosRequest({
+				method: "GET",
+				url: playlistsURL,
+				resKey: "playlist",
+			});
+			console.log({ output });
```
Comment (dev-ashishsethi) [inline src/contexts/videoContext.js:287]: remove console log

**Review by dev-ashishsethi — APPROVED** [review]
looks good

File: src/components/Cards/PlaylistCard/PlaylistCard.css (hunk 1 above)
Comment (PuneetP16) [reply src/components/Cards/PlaylistCard/PlaylistCard.css:6]: will update it

File: src/components/Cards/PlaylistCard/PlaylistCard.jsx (hunk 2 above)
Comment (PuneetP16) [reply src/components/Cards/PlaylistCard/PlaylistCard.jsx:12]: Ok.

File: src/pages/SingleVideo/SingleVideo.js (hunk 5 above)
Comment (PuneetP16) [reply src/pages/SingleVideo/SingleVideo.js:185]: Will look into it.

**Pull request merged**